### DISCUSSION TITLE
Accessibility of API return

### DISCUSSION THREAD

**Discussion by BenjaminOddou** [nuxt/nuxt#15904] [Questions]
Hello folks,

I am fetching data from an api using `server/api/some_api.ts` and I remarked that the response of my $fetch call was stored and accessible through an url `mywebsite/api/some_api`. Is there a way to prevent access to the api ? 

```ts
export default defineEventHandler(async () => {
  const config = useRuntimeConfig()
  const response = await $fetch('API_LINK', {
    method: 'GET',
    headers: {
      Authorization: `Basic ${config.API_KEY}`,
    },
  })
  return response
})
```

Many thanks in advance,

Benjamin

**Comment by mcrapts** [comment]
This is intended behavior. What are you trying to achieve? What would be the purpose of a non-accessible API?

**Comment by TheAlexLichter** [comment]
If you have a static site you could use `nuxt generate` to pre-render all sites and have no API endpoint at all.

Brief pros/cons

Pros: 

* Very fast because static HTML
* No API calls anymore
* Easy hosting

Cons:

* Less flexible because no API endpoint there anymore
* Rebuild necessary when content changes


Other than that, there is no option to "avoid" it AFAIK.

**Reply by BenjaminOddou** [reply]
Thank you for the response @manniL ! 😉 I understand, so for my purpose I need to have a dynamic API endpoint. Maybe that could be an idea for a futur update to be able to hide the API endpoint on client side 🤔

**Reply by TheAlexLichter** [reply]
You can't. If you make a network call, it will always be visible ☺️
